src/main/java/gyro/google/compute/AddressFinder.java

```diff
@@ -0,0 +1,129 @@
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Address;
+import com.google.api.services.compute.model.AddressAggregatedList;
+import com.google.api.services.compute.model.AddressList;
+import com.google.api.services.compute.model.AddressesScopedList;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+
+/**
+ * Query for regional addresses.
+ *
+ * Example find the address filtering on name equal to 'us-east1-test-two' in the region 'us-east1'.
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    addresses: $(external-query google::address { filter: 'name = "us-east1-test-two"', region: 'us-east1' })
+ *
+ */
+@Type("address")
+public class AddressFinder extends GoogleFinder<Compute, Address, AddressResource> {
+
+    private String filter;
+    private String region;
+
+    /**
+     * A filter expression that filters resources listed in the response. The expression must specify the field name, a comparison operator, and the value that you want to use for filtering.
+     */
+    public String getFilter() {
+        return filter;
+    }
+
+    public void setFilter(String filter) {
+        this.filter = filter;
+    }
+
+    /**
+     * Name of the region for this request. Not applicable to global addresses.
+     */
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    @Override
+    protected List<Address> findAllGoogle(Compute client) throws Exception {
+        List<Address> addresses = new ArrayList<>();
+        String pageToken = null;
+
+        do {
+            AddressAggregatedList aggregatedList = client.addresses().aggregatedList(getProjectId())
+                .setPageToken(pageToken)
+                .execute();
+
+            if (aggregatedList.getItems() != null) {
+                aggregatedList.getItems().remove("global");
+                pageToken = aggregatedList.getNextPageToken();
+
+                List<Address> aggregated = aggregatedList.getItems().values().stream()
+                    .map(AddressesScopedList::getAddresses)
+                    .filter(Objects::nonNull)
+                    .flatMap(List::stream)
+                    .collect(Collectors.toList());
+
+                if (!aggregated.isEmpty()) {
+                    addresses.addAll(aggregated);
+                }
+            } else {
+                break;
+            }
+
+        } while (pageToken != null);
+
+        return addresses;
+    }
+
+    @Override
+    protected List<Address> findGoogle(Compute client, Map<String, String> filters) throws Exception {
+        if (filters.containsKey("region")) {
+            List<Address> addresses = new ArrayList<>();
```

Comment:
This line can be taken out of the `if` block, then we will have a single return statement.

Reply:
Done.